packages/xo-server/src/xo-mixins/resource-sets.js

```diff
@@ -174,6 +178,27 @@ export default class {
       set.name = name
     }
     if (subjects) {
+      await Promise.all(
+        difference(set.subjects, subjects).map(async subjectId =>
+          Promise.all(
+            (await this._xo.getAclsForSubject(subjectId)).map(async acl => {
+              try {
+                const object = this._xo.getObject(acl.object)
+                if (object.type === 'VM' && object.resourceSet === id) {
```

Comment:
Why do we need to test the type?

Reply:
To be safe and avoid messing with ACLs of snapshots, templates or any other objects that may have a `resourceSet` field in the future.